opensrp-tasking/src/main/java/org/smartregister/tasking/presenter/TaskingHomePresenter.java

```diff
@@ -0,0 +1,825 @@
+
+        if (getView() != null) {
+            getView().showProgressDialog(R.string.fetching_structures_title, R.string.fetching_structures_message);
+        }
+        taskingHomeInteractor.fetchLocations(currentPlanId, currentOperationalArea);
+    }
+
+    @Override
+    public void onStructuresFetched(JSONObject structuresGeoJson, Feature operationalArea, List<TaskDetails> taskDetailsList, String point, Boolean locationComponentActive) {
+        onStructuresFetched(structuresGeoJson, operationalArea, taskDetailsList);
+    }
+
+    @Override
+    public void onStructuresFetched(JSONObject structuresGeoJson, Feature operationalArea, List<TaskDetails> taskDetailsList) {
+        if (getView() != null) {
+            setChangeMapPosition(drawerPresenter.isChangedCurrentSelection() || (drawerPresenter.isChangedCurrentSelection() && changeMapPosition));
+            drawerPresenter.setChangedCurrentSelection(false);
+            if (structuresGeoJson != null && structuresGeoJson.has(FEATURES)) {
+                featureCollection = FeatureCollection.fromJson(structuresGeoJson.toString());
+                isTasksFiltered = false;
+                if (filterParams != null && !filterParams.getCheckedFilters().isEmpty() && StringUtils.isBlank(searchPhrase)) {
+                    filterFeatureCollection = null;
+                    filterTasks(filterParams);
+                } else if (filterParams != null && !filterParams.getCheckedFilters().isEmpty()) {
+                    searchFeatureCollection = null;
+                    searchTasks(searchPhrase);
+                } else {
+                    getView().setGeoJsonSource(getFeatureCollection(), operationalArea, isChangeMapPosition());
+                }
+                this.operationalArea = operationalArea;
+                if (org.smartregister.util.Utils.isEmptyCollection(getFeatureCollection().features())) {
+                    getView().displayNotification(R.string.fetching_structures_title, R.string.no_structures_found);
+                }
+            } else {
+                getView().displayNotification(R.string.fetching_structures_title,
+                        R.string.fetch_location_and_structures_failed, prefsUtil.getCurrentOperationalArea());
+                try {
+                    if (structuresGeoJson != null) {
+                        structuresGeoJson.put(FEATURES, new JSONArray());
+                        getView().setGeoJsonSource(FeatureCollection.fromJson(structuresGeoJson.toString()), operationalArea, isChangeMapPosition());
+                    }
+                    getView().clearSelectedFeature();
+                    getView().closeCardView(R.id.btn_collapse_spray_card_view);
+                } catch (JSONException e) {
+                    Timber.e("error resetting structures");
+                }
+            }
+            getView().hideProgressDialog();
+
+//            if (taskDetailsList != null && (BuildConfig.BUILD_COUNTRY == Country.ZAMBIA
```

Comment:
I am not sure of the intention when commenting these out. Could you do something about it? Create an issue, todo, remove, fix